mitodl/open-discussions
# Viewing (plaintext) widgets

#### Pre-Flight checklist


#### What are the relevant tickets?
Toward #3694 

#### What's this PR do?
This PR:
1. Renames `pages/admin` to `pages/field-details`, which now holds the routes for viewing + editing field details. I also moved `FieldAvatar`, `FieldMenu` into that directory.
2. Adds a new local package ("yarn workspace") `ol-widgets` for widget-related stuff My plan is to put individual componenets for rendering widgets here.
3. Adds codes for **viewing** a plaintext widget. (Which will become a richtext widget). I'll add editing / deleting / reordering in the next PR.

#### How should this be manually tested?
1. In admin panel, view the WidgetList for a field. (There's a link to it in the Field's details in admin panel)
2. Add a few Markdown widget instances with title + configuration `{ "source": "whatever plaintext you want" }`
3. Check that the widget gets rendered on the field page.

⚠️ When I was adding markdown widget instances in the admin panel, I couldn't edit the configuration field initially. But if you submit the form with an empty config (it will fail) the field becomes editable.

#### Where should the reviewer start?
With the github comments

#### Screenshots

<img width="1158" alt="Screen Shot 2022-08-17 at 1 10 34 PM" src="https://user-images.githubusercontent.com/9010790/185200949-b7a70e83-9aa1-4fb8-ab73-7e003d791fa0.png">
